pkg/io/binaryWriter.go

```diff
@@ -106,23 +106,30 @@ func (w *BinWriter) WriteVarUint(val uint64) {
 		return
 	}
 
+	w.WriteBytes(AppendVarUint(w.u64[:0], val))
+}
+
+// AppendVarUint appends val in varint form to the pre-allocated buffer.
+func AppendVarUint(data []byte, val uint64) []byte {
 	if val < 0xfd {
-		w.WriteB(byte(val))
-		return
+		return append(data, byte(val))
 	}
 	if val < 0xFFFF {
-		w.WriteB(byte(0xfd))
-		w.WriteU16LE(uint16(val))
-		return
+		data = append(data, byte(0xfd))
+		data = append(data, byte(val), byte(val>>8))
+		return data
 	}
 	if val < 0xFFFFFFFF {
-		w.WriteB(byte(0xfe))
-		w.WriteU32LE(uint32(val))
-		return
+		data = append(data, byte(0xfe))
+		data = append(data, byte(val), byte(val>>8), byte(val>>16), byte(val>>24))
+		return data
 	}
 
-	w.WriteB(byte(0xff))
-	w.WriteU64LE(val)
+	data = append(data, byte(0xff))
+	data = append(data,
+		byte(val), byte(val>>8), byte(val>>16), byte(val>>24),
+		byte(val>>32), byte(val>>40), byte(val>>48), byte(val>>56))
```

Comment:
It'd be interesting to compare this with `encoding/binary` in terms of raw speed.

Reply:
I have compared this with old implementation (via `binary.LittleEndian`, but with `WriteU64LE` wrappers removed).
The difference in performance is negligible (big uint64 number):
```
BenchmarkVarint/append-8         	308960446	         3.636 ns/op
BenchmarkVarint/raw-8            	361104601	         3.209 ns/op
```

Reply:
So `raw` is `encoding/binary`?

Reply:
Yes

Reply:
13%, not nothing. What if we `append` 8 bytes and then `PutUint64` into them?

Reply:
Fixed.